worlds/alttp/Shops.py

```diff
@@ -172,6 +172,7 @@ def FillDisabledShopSlots(world):
         shop: Shop = location.parent_region.shop
         location.item = ItemFactory(shop.inventory[location.shop_slot]['item'], location.player)
         location.item_rule = lambda item: item.name == location.item.name and item.player == location.player
+        location.locked = True
```

Comment:
this and

Reply:
They normally get locked right after anyway, but this was due to timing guarenteed by it being in main.py. Since we can now get interjected by other worlds we lock as we place.

Reply:
(It might not be needed, I added this one for safety)